include/whisper.h

```diff
@@ -652,6 +669,68 @@ extern "C" {
     WHISPER_API float whisper_full_get_token_p           (struct whisper_context * ctx, int i_segment, int i_token);
     WHISPER_API float whisper_full_get_token_p_from_state(struct whisper_state * state, int i_segment, int i_token);
 
+    //
+    // Voice Activity Detection (VAD)
+    //
+
+    struct whisper_vad_context;
+    struct whisper_vad_state;
+
+    WHISPER_API struct whisper_vad_params whisper_vad_default_params(void);
+
+    struct whisper_vad_context_params {
+        int   n_threads;  // The number of threads to use for processing.
+        bool  use_gpu;
+        int   gpu_device; // CUDA device
+    };
+
+    WHISPER_API struct whisper_vad_context_params whisper_vad_default_context_params(void);
+
+    WHISPER_API struct whisper_vad_state * whisper_vad_init_state(struct whisper_vad_context * ctx);
+
+    WHISPER_API struct whisper_vad_context * whisper_vad_init_from_file_with_params(
+            const char * path_model,
+            const struct whisper_vad_context_params params);
+
+    WHISPER_API struct whisper_vad_context * whisper_vad_init_from_file_with_params_no_state(
+            const char * path_model,
+            const struct whisper_vad_context_params params);
+
+    WHISPER_API struct whisper_vad_context * whisper_vad_init_with_params_no_state(
+            struct whisper_model_loader * loader,
+            struct whisper_vad_context_params params);
+
+    struct whisper_vad_probs {
+        int     n_probs;
+        float * probs;
+    };
```

Comment:
We can also consider making this struct oblique as well. This makes the API more future-proof.

Reply:
We can also even fully remove this struct and provide the following API:

```c
WHISPER_API int     whisper_vad_n_probs(struct whisper_vad_context * vctx);
WHISPER_API float * whisper_vad_probs  (struct whisper_vad_context * vctx);
```